plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithStringBuilderAppendRangeInspection.kt

```diff
@@ -74,11 +85,25 @@ class ReplaceWithStringBuilderAppendRangeInspection : AbstractKotlinInspection()
             } else if (secondArgAsInt != 0) {
                 thirdArg.replace(psiFactory.createExpressionByPattern("$0 + $1", secondArg, thirdArg))
             }
+
+            if (!isNotNullFirstArg) {
+                firstArg.replace(psiFactory.createExpressionByPattern("$0!!", firstArg))
+            }
         }
 
         private fun KtPsiFactory.createCalleeExpression(functionName: String): KtExpression =
             (createExpression("$functionName()") as KtCallExpression).calleeExpression!!
 
         private fun KtExpression.toIntOrNull(): Int? = safeAs<KtConstantExpression>()?.text?.toIntOrNull()
+
+        private fun KtExpression.isNotNullType(): Boolean {
```

Comment:
I wonder does something like this function already exist in the codebase so we don't have to write it by hand? Seems like it should be pretty common functionality. (I haven't researched this).